rust/xaynet-analytics/src/database/analytics_event/adapter.rs

```diff
@@ -0,0 +1,127 @@
+use anyhow::{anyhow, Error, Result};
+use isar_core::object::{
+    data_type::DataType,
+    isar_object::{IsarObject, Property},
+    object_builder::ObjectBuilder,
+};
+use std::vec::IntoIter;
+
+use crate::database::{
+    common::{FieldProperty, IsarAdapter, RelationalField, Repo, SchemaGenerator},
+    isar::IsarDb,
+    screen_route::data_model::ScreenRoute,
+};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct AnalyticsEventAdapter {
+    pub name: String,
+    pub event_type: i32,
+    pub timestamp: String,
+    pub screen_route_field: Option<RelationalField>,
+}
+
+impl AnalyticsEventAdapter {
+    pub fn new<N: Into<String>>(
+        name: N,
+        event_type: i32,
+        timestamp: String,
+        screen_route_field: Option<RelationalField>,
+    ) -> Self {
+        Self {
+            name: name.into(),
+            event_type,
+            timestamp,
+            screen_route_field,
+        }
+    }
+}
+
+impl<'event> IsarAdapter<'event> for AnalyticsEventAdapter {
+    fn into_field_properties() -> IntoIter<FieldProperty> {
+        // NOTE: properties need to be ordered by type. Properties with the same type need to be ordered alphabetically
+        // https://github.com/isar/isar-core/blob/1ea9f27edfd6e3708daa47ac6a17995b628f31a6/src/schema/collection_schema.rs
+        vec![
+            FieldProperty::new("event_type".to_string(), DataType::Int),
+            FieldProperty::new("name".to_string(), DataType::String),
+            FieldProperty::new("screen_route_field".to_string(), DataType::String),
+            FieldProperty::new("timestamp".to_string(), DataType::String),
+        ]
+        .into_iter()
+    }
+
+    fn write_with_object_builder(&self, object_builder: &mut ObjectBuilder) {
+        let screen_route_field: Option<&str> = if let Some(field) = &self.screen_route_field {
```

Comment:
This can be written as: `self.screen_route_field.as_ref().map(|field| field.value.as_str())`.

Reply:
looks good, will change